### PR TITLE
Add support for running Linux desktop tests on CI

### PR DESCRIPTION
**Pull request by stuartmorgan**
## Description

Adds support for building and driving e2e tests for Linux (as a target).

## Related Issues

https://github.com/flutter/flutter/issues/55327

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [ ] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [ ] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -95,6 +95,13 @@ task:
         - fi
         - export CIRRUS_CHANGE_MESSAGE=`cat /tmp/cirrus_change_message.txt`
         - export CIRRUS_COMMIT_MESSAGE=`cat /tmp/cirrus_commit_message.txt`
+    - name: build-linux+drive-examples
+      install_script:
+        - flutter config --enable-linux-desktop
+      build_script:
```
Comment (cyanglaz) [inline .cirrus.yml:101]: Should we run the tests on stable, too? 
Or it is in the future plan, can we add a TODO here?

Comment (stuartmorgan) [reply .cirrus.yml:101]: Like web, stable isn't supported for Linux. I'll add a TODO.

**Comment by stuartmorgan** [comment]
> CI seems failing

I'm assuming it's infra failure since it said the agent stopped responding. Trying a re-run, and will escalate to infra channel if it keeps failing.

**Comment by stuartmorgan** [comment]
The failing web tests are using Chrome 81, while the passing master tests are using Chrome 79; my best guess at this point is that the web tests are broken with newer Chrome and my PR is hitting what is essentially submarine bustage because it's rebuilding the Docker image.

**Comment by stuartmorgan** [comment]
@hterkelsen It looks like you added the code for this in https://github.com/flutter/plugins/pull/2307; is there a way we can structure this so that we control what version of Chrome we are using for e2e tests instead of random PRs touching the Dockerfile having the side-effect of updating Chrome?

**Comment by stuartmorgan** [comment]
Landing with a modified approach that uses a separate Dockerfile and separate top-level task for the Linux desktop tests, so that progress on Linux isn't gated on the web issue.

At some point when web tests work with the latest Chrome, or the Dockerfile is changed not to rely on a floating Chrome version, they could be re-combined.

**Pull request merged**